### ISSUE TITLE
Initial questions

### ISSUE THREAD

**Issue by jeff-h** [oleganza/gitbox#1]
Hi @oleganza — firstly thanks so much for sharing this!

So I've had a crack at compiling this in Xcode 16.0. Went pretty well; I had to:

* have Xcode upgrade each `xib`
* rip out all `dispatch_retain` and `dispatch_release` (I'm not super familiar with this, I assume ARC will just do its thing?)
* compiled `libgit2` — I had to tell it I want x86 in your `update_libgit2.sh` script: `cmake .. -DBUILD_SHARED_LIBS=OFF -DCMAKE_OSX_ARCHITECTURES=x86_64`
* pulled in the submodules

On first launch it's currently failing due to exceeding its open file count which is weird. I did have about 20–30 repos in my App-store version of Gitbox, which this build picked up on, and with those removed it launches fine.

Some questions:

* would you be happy to build this only for Apple Silicon from here on?
* are you happy with a minimum macOS version of 11 (Big Sur)? I know that's pretty arbitrary, and is in a sense a wasted opportunity to have it work for others on older systems, but it might be nice to be able to use newer APIs over time?
* are you happy for me to rip out all the non-MAS licensing stuff? Should I leave in the MAS stuff?
* what's the need for the bespoke openssl? I haven't complied it using `update_openssl.sh` yet (had a quick try which failed).

And overall, are you happy to accept PRs here, or should I just move ahead in my own fork?

**Comment by oleganza** [comment]
Thanks for looking into building Gitbox! 

Quick answers to your questions:
1 - apple silicon sounds fine
2 - bigsur ok
3 - let's remove MAS - ideally we'd re-ship on App Store free of charge
4 - i think there are some historical reasons for openssl being what it is. The story of OpenSSL was rapidly changing after Snowden in 2013 and with all the cryptocurrency explosion in 2014-2017.

Also: i think ARC did not support Dispatch back then, but later more of the APIs were made ARC-compatible. Maybe ripping out dispatch_retain/release is safe, maybe not - i haven't checked lately.

**Comment by oleganza** [comment]
@jeff-h i'm happy to accept PRs here.

**Comment by tillsc** [comment]
Hey, just wanted to leave a note that there's an active fork at https://github.com/tillsc/gitbox that currently builds on modern macOS.

It builds on top of the work done by https://github.com/larssonphilip (updated submodules, fixed encoding issues, macOS compatibility). Without that foundation it would have been much harder to get going. The additional changes on top are: a build.sh script, removing expired code signing identities, fixing the opendiff retry logic to handle xcode-select errors, and removing the license/DRM infrastructure so it can be built freely. 
 
No PR yet (partly because I'm not sure why @larssonphilip's changes haven't been PRed upstream), but wanted to share in case anyone else is trying to get gitbox to compile today.

**Comment by larssonphilip** [comment]
I can definitely PR my changes if needed :) 

I made the changes a while ago and then decided to make my own "gitbox" from scratch in Swift instead! So I haven't touched it in forever

**Comment by tillsc** [comment]
Nice, that would be great!

If you’re okay with it, I can also include your changes directly in a PR on my side so everything ends up upstream in one go. Saves some coordination and gets things merged faster.

See: https://github.com/oleganza/gitbox/compare/master...tillsc:gitbox:master

Let me know if that works for you

**Comment by jeff-h** [comment]
This sounds great. Will you guys PR your work into this repo? 

I also have a functional build of Gitbox but mine still has a few rough edges. I never got around to converting it to a PR as I wasn't really certain it would be useful as one huge PR.

@oleganza Do you have an active Apple developer account? If not, I wonder if we could create an automated build using mine. Would that be something you'd consider? I think a precompiled release would greatly increase the likelihood people will find this useful.

**Comment by larssonphilip** [comment]
> Nice, that would be great!
> 
> If you’re okay with it, I can also include your changes directly in a PR on my side so everything ends up upstream in one go. Saves some coordination and gets things merged faster.
> 
> See: https://github.com/oleganza/gitbox/compare/master...tillsc:gitbox:master
> 
> Let me know if that works for you

This will make you a contributor to the project, but not me, correct?

**Comment by tillsc** [comment]
> This will make you a contributor to the project, but not me, correct?

No, your commits are part of the PR, so you’ll be credited too, as long as the PR is merged without squashing away your authorship.

**Comment by larssonphilip** [comment]
> > This will make you a contributor to the project, but not me, correct?
> 
> No, your commits are part of the PR, so you’ll be credited too, as long as the PR is merged without squashing away your authorship.

Alright, then feel free to do the PR! :) 

I love that people are starting to contribute to this! Gitbox was my favorite git tool when it was maintained!

**Comment by tillsc** [comment]
See #3

**Comment by jeff-h** [comment]
The activity in this thread (so good to see life in this project again!) has prompted me to take another look at my codebase  work from late 2024.

Compared with [#3](https://github.com/oleganza/gitbox/pull/3) I've taken a somewhat different approach in places, and my fork includes some additional changes that might be useful:

**Bug fixes:**

- Fixed a missing comma in GBStage.m where @"--ignore-removal" @"." was being concatenated into a single malformed argument
- Fixed URL construction in GBRepository.m (proper fileURLWithPath:relativeToURL: instead of manual percent-encoding)
- Replaced deprecated dispatch_get_current_queue() in OATask.m (was deprecated since macOS 10.9 and could cause deadlocks)
- Removed incorrect NSCaseInsensitiveSearch in error string matching (GBChange.m)
- Removed spurious stringByRemovingPercentEncoding calls on file paths (GBChangeCell.m)

**Modernisation:**

- Enabled ARC project-wide (with corresponding manual retain/release cleanup)
- Removed the OpenSSL dependency entirely
- Upgraded bundled Git from 1.8.4 → 2.50.1
- Upgraded Sparkle from 1.5 Beta 6 → 2.6.4
- Converted 4 xibs from legacy IB Archive format to modern XIB format
- Repointed submodules to oleganza's original repos
- General cleanup (removed stale debug logging, tidied toolbar item management)
 
Apologies it's taken me so long to share it. Happy for @oleganza to determine the best path forward. I realise these monolithic PRs are the stuff of nightmares for a maintainer, so I'm open to alternative ideas on how to approach this.

ps I'm also building signed releases now over at https://github.com/jeff-h/gitbox/releases, just so there's a usable build available for those who don't want to build it themselves.
